CrewChiefV4/UserInterface/PropertiesForm.cs

```diff
@@ -179,5 +199,203 @@ private void properties_FormClosing(object sender, FormClosingEventArgs e)
                 }
             }           
         }
+
+        private void textBox1_TextChanged(object sender, EventArgs e)
+        {
+            this.nextPrefsRefreshAttemptTime = DateTime.Now.Add(AUTO_SEARCH_DELAY_SPAN);
+
+            if (this.textBox1.Text == DEFAULT_SEARCH_TEXT)
+                return;
+
+            if (this.searchTimer == null)
+            {
+                this.searchTimer = new Timer();
+                this.searchTimer.Interval = 100;
+                this.searchTimer.Tick += SearchTimer_Tick;
+                this.searchTimer.Start();
+            }
+        }
+
+        private void SearchTimer_Tick(object sender, EventArgs e)
+        {
+            if (DateTime.Now < this.nextPrefsRefreshAttemptTime)
+                return;
+
+            var text = this.textBox1.Text;
+            if (text == DEFAULT_SEARCH_TEXT)
+            {
+                this.searchTextPrev = text;
+                return;
+            }
+
+            if (text != this.searchTextPrev)
+            {
+                // This is the case on entering the text box
+                if (string.IsNullOrWhiteSpace(text) && this.searchTextPrev != DEFAULT_SEARCH_TEXT)
+                    this.PopulatePrefsFiltered(text);
+                // General case, new filter.
+                else if (!string.IsNullOrWhiteSpace(text))
+                    this.PopulatePrefsFiltered(text);
```

Comment:
Yes.  This is the type of feature where there are many special cases.  Entering/exiting control etc etc etc.  If you find a bug, let me know.